api/v1alpha1/nutanix_node_types.go

```diff
@@ -60,6 +60,25 @@ type NutanixMachineDetails struct {
 	// The project must already be present in the Prism Central.
 	// +kubebuilder:validation:Optional
 	Project *NutanixResourceIdentifier `json:"project,omitempty"`
+
+	// List of GPU devices that need to be added to the machines.
+	// +kubebuilder:validation:Optional
+	GPUs []NutanixGPU `json:"gpus,omitempty"`
+}
+
+type NutanixGPU struct {
+	// Type is the identifier type to use for this resource.
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:Enum:=deviceID;name
+	Type string `json:"type"`
```

Comment:
Did you consider this?

```suggestion
	Type NutanixGPUIdentifierType `json:"type"`
```

If you use that, please add an alias at the end of this file:
```
type NutanixGPUIdentifierType capxv1.NutanixGPUIdentifierType
```

Reply:
i suppose we could add this but since we're not doing anything with the field other than copying it, i don't think its necessary at this point.

Reply:
That's the convention we've been following for similar fields :shrug:

Reply:
boom done.